python/cudf/cudf/io/csv.py

```diff
@@ -114,17 +113,25 @@ def to_csv(
                 columns = columns.copy()
                 columns.insert(0, df.index.name)
         df = df.reset_index()
+
+    if columns is not None:
+        try:
+            df = df[columns]
```

Comment:
So this is done instead of passing a columns parameter. To me it looks like it modifies the dataframe that was passed into the `to_csv` call (unless there's some Python quirk happening here). I don't think we should be modifying an input parameter.

Reply:
`columns` are the names of the columns present in dataframe that needs to be written. So, we don't need other set of columns. And here we are not modifying input, it just creates another shallow copy of selected columns. And we also check if names provided in `columns` is present in dataframe or not.

Reply:
From an offline discussion: modifying a parameter affects the object outside the function, but reassigning the parameter object completely does not. Looks like semantics are the same as passing a pointer in C++.
So this is fine.